lib/src/components/board/board.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
```

Comment:
what does `factory` do here? Without the `factory`, is it not just a named initializer?

Reply:
Oh cool. This was more of a question for me to understand what `factory` means hahaha but glad to have helped make this more idiomatic

Reply:
upon further inspection, the use of factory does provide the benefit of being able to place logic inside the constructor and return a specific object. I'll prefer to keep it a factory so I can continue to map the pb model inside it. Let me know if you feel strongly otherwise.